### PR TITLE
Add a builder for LimitReductions

### PR DESCRIPTION
**Pull request by olperr1**
**Please check if the PR fulfills these requirements**

- [X] The commit message follows our guidelines
- [X] Tests for the changes have been added (for bug fixes / features)
- [X] Docs have been added / updated (for bug fixes / features)

**Does this PR already have an issue describing the problem?**

No

**What kind of change does this PR introduce?**

Feature + doc update

**What is the current behavior?**

When creating a `LimitReduction` with either a contingency context or a criterion list, every characteristics of a `LimitReduction` must be given as parameter for the constructor. This forces the user to give empty lists to the constructor for the unneeded parameters. 

**What is the new behavior (if this is a feature change)?**
- Two small constructors are now present for simple cases:
  - `LimitReduction(LimitType limitType, double value)`;
  - `LimitReduction(LimitType limitType, double value, boolean monitoringOnly)`.
- For more complex cases, a builder was added.

**Does this PR introduce a breaking change or deprecate an API?**
- [X] Yes
- [ ] No

**If yes, please check if the following requirements are fulfilled**

- [X] The *Breaking Change* or *Deprecated* label has been added
- [ ] The migration steps are described in the following section

**What changes might users need to make in their application due to this PR? (migration steps)**

When you used the constructor with all the parameters to create a "LimitReduction", you should use the builder.
Default values for the characteristics are:
- `monitoringOnly`: false;
- `contingencyContext`: `ContingencyContext.all()`;
- `networkElementCriteria`: `Collections.emptyList()`;
- `limitDurationCriteria`: `Collections.emptyList()`.

Thus, you can omit setting one of the characteristics if you used its default value.

For instance:

```java
       LimitReduction limitReduction = new LimitReduction(LimitType.APPARENT_POWER, 0.5, false,
                ContingencyContext.all(),
                Collections.emptyList(),
                List.of(durationCriterion1, durationCriterion2));
```

could be rewritten in

```java
       LimitReduction limitReduction = LimitReduction.builder(LimitType.APPARENT_POWER, 0.5)
                .withLimitDurationCriteria(durationCriterion1, durationCriterion2)
                .build();
```

Note:
- `monitoringOnly`, `networkElementCriteria` and `contingencyContext` are not specified as their default values are used;
- the `with...Criteria(...)` methods take either a `List` or varArgs. You could thus omit the encapsulating `List.of()` for ease of use. But in either case, the previously definied criteria are replaced by the given ones. 

**Other information**:

### CODE REVIEW

File: security-analysis/security-analysis-api/src/main/java/com/powsybl/security/json/limitreduction/LimitReductionDeserializer.java [hunk 1]
```diff
@@ -76,9 +75,17 @@ public LimitReduction deserialize(JsonParser parser, DeserializationContext dese
                 }
             }
         });
-        return new LimitReduction(context.limitType, context.value, context.monitoringOnly,
-                context.contingencyContext,
-                context.networkElementCriteria != null ? context.networkElementCriteria : Collections.emptyList(),
-                context.durationCriteria != null ? context.durationCriteria : Collections.emptyList());
+        LimitReduction.Builder builder = LimitReduction.builder(context.limitType, context.value)
+                .withMonitoringOnly(context.monitoringOnly);
+        if (context.contingencyContext != null) {
```
Comment (annetill) [inline security-analysis/security-analysis-api/src/main/java/com/powsybl/security/json/limitreduction/LimitReductionDeserializer.java:80]: What is the meaning of a null contingency context?

Comment (olperr1) [reply security-analysis/security-analysis-api/src/main/java/com/powsybl/security/json/limitreduction/LimitReductionDeserializer.java:80]: We are in the deserializer here, the contingency context may not have been specified in the JSON file to import.

Comment (annetill) [reply security-analysis/security-analysis-api/src/main/java/com/powsybl/security/json/limitreduction/LimitReductionDeserializer.java:80]: Why do we have these two ways? Null and ALL ?

Comment (olperr1) [reply security-analysis/security-analysis-api/src/main/java/com/powsybl/security/json/limitreduction/LimitReductionDeserializer.java:80]: Ok, I will set the ContingencyContext as mandatory in the JSON file.

**Review by annetill — APPROVED** [review]

File: security-analysis/security-analysis-api/src/main/java/com/powsybl/security/json/limitreduction/LimitReductionDeserializer.java (hunk 1 above)
Comment (olperr1) [reply security-analysis/security-analysis-api/src/main/java/com/powsybl/security/json/limitreduction/LimitReductionDeserializer.java]: The ContingencyContext is now mandatory in the JSON files to import.

**Pull request merged**